feat: add obsidian ingestion support

0xrushi opened the pull request:
Needs neo4j to be enabled, thats disabled by default 

<img width="1032" height="350" alt="image" src="https://github.com/user-attachments/assets/74516d79-0b9c-4e36-8314-da72e3df768f" />


## new note 
<img width="1551" height="607" alt="image" src="https://github.com/user-attachments/assets/f7040040-8491-40e1-aba1-95c2bb6cde78" />

## client connected to chronicle neo4j
<img width="933" height="138" alt="image" src="https://github.com/user-attachments/assets/653566e3-4900-4d1b-8922-25d1cf15a2b4" />

### enabled obsidian memory, asked a query, then disabled and asked same query 
<img width="1262" height="869" alt="sarah1" src="https://github.com/user-attachments/assets/74dc2722-0d23-46e3-afba-6f2d8aca9145" />

### new chat with obsidian memory disabled
<img width="1043" height="871" alt="sarah2" src="https://github.com/user-attachments/assets/df35c355-9598-44ac-88bc-905965538034" />



<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Obsidian vault import: upload ZIPs, start ingestion, and view real‑time progress/status in UI
  * Chat option "Include Obsidian Memory" to augment responses with vault content
  * New Obsidian import UI/component for authenticated/admin users

* **Enhancements**
  * Admin‑only gating and clearer error messages for memory/ingest workflows
  * Backend integration for Neo4j‑based storage and improved embedding/search

* **Tests**
  * Added unit tests covering Obsidian ingestion and search flows

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

0xrushi committed:
obsidian support

0xrushi committed:
neo4j comment

0xrushi committed:
cleanup code

0xrushi committed:
Merge github.com:0xrushi/friend-lite into feat/add-obsidian

0xrushi committed:
unused line

0xrushi committed:
unused line

0xrushi committed:
Fix MemoryEntry object usage in chat service

0xrushi committed:
comment

0xrushi committed:
feat(obsidian): add obsidian memory search integration to chat

0xrushi committed:
unit test

0xrushi commented on backends/advanced/src/advanced_omi_backend/memory_config_loader.py:
Single config has memory and llm models

0xrushi committed:
use rq

0xrushi committed:
Merge branch 'feat/add-obsidian' of github.com:0xrushi/friend-lite into feat/add-obsidian

thestumonkey commented on backends/advanced/src/advanced_omi_backend/services/obsidian_job_tracker.py:
This looks like it should live in utils/job_utils or queue_controller (which should maybe be job controller).  this seems like an interface to redis and only the job prefix has anything to do with obsidian service

thestumonkey replied:
we also have a job base class which is used for processing jobs on redis using RQ.  Please use this for processing jobs, as this is handled natively with RQ.

thestumonkey replied:
also functions in queue_routes that should provide what you need.  This will then hook it into the queue management page so we can see progress.

thestumonkey commented on backends/advanced/src/advanced_omi_backend/routers/modules/obsidian_routes.py:
```diff
@@ -0,0 +1,167 @@
+
+
+@router.post("/upload_zip")
+async def upload_obsidian_zip(
```
use upload_files_async in upload_files.py (which should be moved to utils)

thestumonkey commented on backends/advanced/src/advanced_omi_backend/routers/modules/obsidian_routes.py:
```diff
@@ -0,0 +1,167 @@
+
+
+@router.get("/status")
+async def get_status(job_id: str, current_user: User = Depends(current_active_user)):
```
This should be hooked into the queue system and get a queue or job status

thestumonkey commented on backends/advanced/src/advanced_omi_backend/routers/modules/obsidian_routes.py:
```diff
@@ -0,0 +1,167 @@
+
+
+@router.post("/cancel")
+async def cancel_job(job_id: str = Body(..., embed=True), current_user: User = Depends(current_superuser)):
```
part of jobs ecossytem, jobs can be cancelled using queue routes

thestumonkey commented on backends/advanced/src/advanced_omi_backend/routers/modules/obsidian_routes.py:
```diff
@@ -0,0 +1,167 @@
+    job_dir.mkdir(parents=True, exist_ok=True)
+    zip_path = job_dir / "vault.zip"
+    extract_dir = job_dir / "vault"
+    try:
```
worth making a extract zip function in utils, we will need this for other things I suspect

thestumonkey commented on backends/advanced/src/advanced_omi_backend/routers/modules/obsidian_routes.py:
```diff
@@ -0,0 +1,167 @@
+    if not vault_path or not os.path.exists(vault_path):
+        raise HTTPException(status_code=400, detail="Vault path not available for this job")
+
+    rq_job = default_queue.enqueue(
```
Best practice to give it a queue name.  Workers can then be assigned to 1 or more queues to process

thestumonkey commented on backends/advanced/src/advanced_omi_backend/routers/modules/obsidian_routes.py:
```diff
@@ -0,0 +1,167 @@
+        description=f"Obsidian ingestion for job {job_id}",
+    )
+
+    update_job_state(
```
Not needed as queues are dealt with by RQ

thestumonkey commented on backends/advanced/src/advanced_omi_backend/routers/modules/system_routes.py:
```diff
@@ -72,35 +72,31 @@ async def get_speaker_service_status(current_user: User = Depends(current_superu
 async def update_memory_config_raw(
-    config_yaml: str = Body(..., embed=True),
-    current_user: User = Depends(current_superuser)
+    config_yaml: str = Body(..., media_type="text/plain"),
```
Are we just getting this as plain text and not parsing yaml?  If this is the case, shouldn't memory config be a text file?  we can use pyYAML or ruamel https://lyz-code.github.io/blue-book/coding/python/ruamel_yaml/.

Looks like this function is pre-existing so can add as #TODO add proper yml parsing and address later during fiurther config changes
/

AnkushMalaker replied:
Yeah this is true, we need to upgrade to ruamel, which I think I saw in develop branch. This is fine for this PR I think though as its a widespread change we need to address across the codebase?

thestumonkey commented on backends/advanced/src/advanced_omi_backend/services/obsidian_service.py:
```diff
@@ -0,0 +1,462 @@
+    text: str
+    embedding: List[float]
+
+def load_env_file(filepath: Path) -> dict[str, str]:
```
we have env loaders in upload_audio/load_env_file, also in root/scripts/lib/env_utils and wizard.py
Feel free to consolidate, but not need to have env loading in the service

thestumonkey commented on backends/advanced/src/advanced_omi_backend/services/obsidian_service.py:
```diff
@@ -0,0 +1,462 @@
+        config_data = load_root_config()
+
+        # Helper to get model config
+        def get_model_config(model_role: str):
```
consider moving to a models utils file as we can use this function for other llm services, this isn't specific to obsidian

thestumonkey commented on backends/advanced/src/advanced_omi_backend/services/obsidian_service.py:
As far as I can see, this service goes through the markdown files creates an LLM and writes directly to NEO4J.

This can be achieved using the current infra by just passing the text to the memory router add_memory
(which writes to neo4J using openmemory)

If there are specific obsidian characteristics, then I would make a model for obsidianmemory/memory and then use a custom prompt to process it using memory controller.

thestumonkey commented on backends/advanced/src/advanced_omi_backend/services/obsidian_service.py:
```diff
@@ -0,0 +1,462 @@
+            """
+            session.run(index_query)
+
+    def get_embedding(self, text: str) -> Optional[List[float]]:
```
see services/memory/providers/llm_provider

thestumonkey commented on extras/obsidian-injest/config.py:
why is this in extras? duplicates existing fucntions

thestumonkey commented on extras/obsidian-injest/search_query.py:
why is this in exteras

thestumonkey reviewed (changes requested):
This PR has a lot of duplication instead of using standard functions we have already.

1:  jobs.  all jobs should be processed by queues and status shown on the queue management page using queue controller

2: upload (small issue):  we should leverage existing upload functions we have to upload gdrive or audio to extend to upload zip.

3: service and processing.  We currently extract memories from conversations, pass this to openmemory which uses neo4j as graph memory for storing inference and memories.  The obsidian service is writing direct to neo4j, which creates multiple memory banks for chat to leverage instead of a single one.  

4: there are some config changes here and use of llm models.  this should be linked into the existing llm provider class

We should use openmemory to process the obsidian files directly, and use a type to send to it. (conversation or note).

Otherwise, if we don't want to do this, we should connect to neo4j via a proper client which isn't part of obsidian service.

0xrushi committed:
neo4j service

0xrushi committed:
typefix

0xrushi committed:
test fix

0xrushi committed:
Merge branch 'dev' into feat/add-obsidian

0xrushi committed:
cleanup

0xrushi committed:
Merge branch 'feat/add-obsidian' of github.com:0xrushi/friend-lite into feat/add-obsidian

0xrushi committed:
Merge branch 'dev' into feat/add-obsidian

0xrushi committed:
cleanup

0xrushi committed:
Merge branch 'feat/add-obsidian' of github.com:0xrushi/friend-lite into feat/add-obsidian

AnkushMalaker commented:
> Can you also provide some obsidian test data and put into tests/test_assets ?

@thestumonkey https://github.com/s-blu/obsidian_dataview_example_vault
This is the example vault used
And the edits made in the screenshot, ie, add a fake birthday name, then ingest the vault and ask the specific question

0xrushi committed:
version changes

0xrushi committed:
profile

AnkushMalaker committed:
remove unused imports

AnkushMalaker committed:
Refactor memory configuration validation endpoints

- Removed the deprecated `validate_memory_config_raw` endpoint and replaced it with a new endpoint that accepts plain text for validation.
- Updated the existing `validate_memory_config` endpoint to clarify that it now accepts JSON input.
- Adjusted the API call in the frontend to point to the new validation endpoint.

AnkushMalaker committed:
Refactor health check model configuration loading

- Updated the health check function to load model configuration from the models registry instead of the root config.
- Improved error handling by logging warnings when model configuration loading fails.

0xrushi committed:
Merge pull request #4 from chronicler-ai/feat/add-obsidian-3

Feat/add obsidian 3

AnkushMalaker committed:
Merge branch 'dev' into feat/add-obsidian
